checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.google.common.base.Preconditions;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+/**
+ * Some Hive queries to test the integration of Hive with Alluxio.
+ * Supports two options: Alluxio is used as storage of Hive tables
+ * and Alluxio is configured as Hive default filesytem.
+ *
+ * This checker requires a running Hive cluster and a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hive queries.
+ */
+@SuppressFBWarnings("SQL_PREPARED_STATEMENT_GENERATED_FROM_NONCONSTANT_STRING")
+public class HiveIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIntegrationChecker.class);
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "Please set HIVE_AUX_JARS_PATH either in shell or in conf/hive-env.sh "
+      + "before starting hiveserver2.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hive-with-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl property "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Debugging-Guide.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  @Parameter(names = {"-alluxioUrl"}, description = "the alluxio cluster url of form "
```

Comment:
let's later have a separate PR to use `--arg` if that is what we decided, in order to keep the code base consistent.
But perhaps not in this PR.

Reply:
I maintain -arg here and later change in other PR if decide.
Consistent is great!